pkg/cloud_provider/storage/storage.go

```diff
@@ -291,3 +301,33 @@ func ParseErrCode(err error) codes.Code {
 
 	return code
 }
+
+// UploadGcsObject uploads a local file to a specified GCS bucket and object.
+// Handles gzip compression if requested.
+func (service *gcsService) UploadGcsObject(ctx context.Context, filePathToUpload, bucketName, objectName string, zbEnabled bool) error {
```

Comment:
nit: this should be UploadGCSObject as GCS is an initialism  go/go-style/decisions#initialisms